### ISSUE TITLE
More flexibility with GridCol

### ISSUE THREAD

**Issue by stevesims** [UKHomeOffice/govuk-react#377]
Currently `GridCol` supports flags for 1/4, 1/3, 1/2 and 2/3 width columns.

By default a column is 100% wide if a width flag is not provided.  Therefore, as there is not a flag to support it, it is not possible to make a 3/4 width column.  It is therefore suggested that we should add support for a `columnThreeQuarters` flag.

Additionally there is no "auto" width column to make use of the remaining flex area in a `GridRow`.  To provide this we could change the default column width be `auto` with `flex: 1` set, and have the explicit widths set `flex: 0`.  (This change would arguably negate the need for an explicit `columnThreeQuarters`.)  Alternatively we could provide an explicit `columnAuto` flag to set `width: auto; flex: 1;`.

It is also not possible to do columns of 1/6th or 5/6th (both of which would be useful in our application). 
 Adding support for `columnOneSixth` and `columnFiveSixths` would also be cool, but as those are a less common requirement and may not form part of the main GDS styling options we could instead add support for those in our other component library.

**Comment by penx** [comment]
I was hesitant to include any kind of grid in govuk-react.

The purpose of the govuk-react grid is to:

- provide any grid support needed internally by components
- mirror what is offered by govuk-frontend (or previous projects)
- provide a grid that is useful enough for examples

It is expected that a parent project can choose to use their own grid system and these components could live within it.

I don't think we should be going down the route of building a feature rich grid system, when there are existing React grid systems that we could adopt instead.

That said, small changes like this aren't the end of the world if the general feeling is they would be useful.

**Comment by penx** [comment]
Related, previous decision on this:

https://github.com/UKHomeOffice/govuk-react/issues/61#issuecomment-368578006

**Comment by penx** [comment]
My vote would be to review existing grid systems and recommend one (or several) in the readme.

**Comment by penx** [comment]
https://github.com/emotion-js/grid-emotion

=> 

https://jxnblk.com/grid-styled/

**Comment by Loque-** [comment]
We've discussed the requirement for having `columnOneSixth` and `columnFiveSixths` concluding that it is probably not required for the project as the UI design is likely to change and no longer require these.

I've implemented the `flex-grow` adjustment, and any `gridCol` that doesn't have one of the designated properties will expand to fill with auto width.

I personally like the change and also agree that the grid component should only facilitate the layouts supported by govuk-frontend.
